### DISCUSSION TITLE
Open Settings Dialog with TextInput

### DISCUSSION THREAD

**Discussion by Korbi-code** [beeware/toga#2170] [Q&A]
Hello,

I'm currently figuring out what the best solution is to open a "Settings" menu that will show up, similar to the info_dialog.

It should have an input field where the user can type in an API key needed by the main application. 

I have some ideas, but I want to know how this is supposed to be done or how others have done something like this.

Thanks in advance

**Comment by freakboy3742** [comment]
Right now, Toga doesn't have a built-in mechanism for displaying/editing settings - it's been on our wish list for a long time (#90). 

The immediate workaround is to create a `Toga.Window` to serve as the dialog, modify the default "settings/preferences" command to display that dialog when requested (or install a new command of your own), and store settings as user data using the `toga.App.paths.config` file prefix.

**Comment by Korbi-code** [comment]
@freakboy3742 is this still the best way to do ?

**Reply by freakboy3742** [reply]
With the current set of features of Toga - yes.
